jwst/associations/lib/rules_level2_base.py

```diff
@@ -133,59 +127,76 @@ def get_exposure_type(self, item, default='science'):
         default : str or None
             The default exposure type.
             If None, routine will raise LookupError
+
         Returns
         -------
         exposure_type
             Always what is defined as `default`
         """
-        self.original_exposure_type = super(DMSLevel2bBase, self).get_exposure_type(item, default=default)
-        if self.original_exposure_type == 'psf':
+        self.original_exposure_type = super(DMSLevel2bBase, self).get_exposure_type(
+            item, default=default
+        )
+        if self.original_exposure_type == "psf":
             return default
         return self.original_exposure_type
 
     def members_by_type(self, member_type):
-        """Get list of members by their exposure type"""
+        """
+        Get list of members by their exposure type.
+
+        Returns
+        -------
+        list
+            List of members.
+        """
         member_type = member_type.lower()
         try:
-            members = self.current_product['members']
+            members = self.current_product["members"]
         except KeyError:
             result = []
         else:
-            result = [
-                member
-                for member in members
-                if member_type == member['exptype'].lower()
-            ]
+            result = [member for member in members if member_type == member["exptype"].lower()]
 
         return result
 
     def has_science(self):
-        """Does association have a science member
+        """
+        Check association for a science member.
 
+        Returns
         -------
         bool
             True if it does.
         """
-        limit_reached = len(self.members_by_type('science')) >= 1
+        limit_reached = len(self.members_by_type("science")) >= 1
         return limit_reached
 
     def dms_product_name(self):
-        """Define product name."""
+        """
+        Define product name.
+
+        Returns
+        -------
+        str
+            The product name.
+        """
         try:
-            science = self.members_by_type('science')[0]
+            science = self.members_by_type("science")[0]
         except IndexError:
             return PRODUCT_NAME_DEFAULT
 
         try:
-            science_path, ext = splitext(science['expname'])
+            path_expname = Path(science["expname"])
+            science_path = path_expname.parent / path_expname.stem
         except Exception:
             return PRODUCT_NAME_DEFAULT
 
-        no_suffix_path, separator = remove_suffix(science_path)
+        no_suffix_path, separator = remove_suffix(str(science_path))
```

Comment:
I see this function lives in `lib` so perhaps outside the scope, but would it make sense to instead allow `remove_suffix` to take either a string or a Path?

Reply:
Agreed on out of scope - we could have a separate PR that updates the method and checks usage across all steps/modules. I think it's best to leave that out of this PR if possible, though.